File: Test/AdventureWorksFunctionalModel/Sales/CustomerDashboard.cs
using AW.Functions;

namespace AW.Types;

    [ViewModel(typeof(CustomerDashboard_Functions))]
    public class CustomerDashboard {
        [Hidden]
        public virtual Customer Root { get; init; }

        public string Name => $"{(Root.IsIndividual() ? Root.Person : Root.Store?.Name)}";

        //Empty field, not - to test that fields are not editable in a VM
        public string Comments { get; init; } = "";

        [TableView(true, "OrderDate", "TotalDue", "Status")]
        public IList<SalesOrderHeader> RecentOrders(IContext context) =>
            Root.RecentOrders(context).Take(5).ToList();

        public override string ToString() => $"{Name} - Dashboard";
    }

File: Test/AdventureWorksLegacyModelVB/Sales/CustomerDashboard.vb


Namespace AW.Types

	'TODO
	'<ViewModel(GetType(CustomerDashboard_Functions))>
	Partial Public Class CustomerDashboard
		<Hidden>
		Public Overridable Property Root() As Customer

		Public ReadOnly Property Name() As String
			Get
				Return Nothing '$"{(If(Root.IsIndividual(), Root.Person, Root.Store?.Name))}"
			End Get
		End Property

		'Empty field, not - to test that fields are not editable in a VM
		Public Property Comments() As String = ""

		Public Overrides Function ToString() As String
			Return $"{Name} - Dashboard"
		End Function
	End Class
End Namespace